File: android/app/src/main/java/jaclib/memory/NativeBuffer.java
package jaclib.memory;


public class NativeBuffer implements Buffer, Source {

    public long b;

    public int a = -1;

    @Override
    public void a(byte[] arg0, int arg1, int arg2, int arg3) {
        if (arg2 < 0 | arg0 == null | this.b == 0L | arg1 < 0 | arg0.length < arg1 + arg3 | arg2 + arg3 > this.a) {
            throw new RuntimeException();
        }
        this.put(this.b, arg0, arg1, arg2, arg3);
    }

    protected final void a(long arg0, int arg1) {
        this.b = arg0;
        this.a = arg1;
    }

    @Override
    public final long getAddress() {
        return this.b;
    }

    public void put(long arg0, byte[] arg1, int arg2, int arg3, int arg4) {}

    public void get(long arg0, byte[] arg1, int arg2, int arg3, int arg4) {}

    @Override
    public final int getSize() {
        return this.a;
    }
}

File: src/jvmMain/kotlin/jaclib/memory/NativeBuffer.kt
package jaclib.memory


open actual class NativeBuffer : Buffer, Source {
    /*private*/ actual var a: Long = 0

    /*private*/ actual var b = -1

    actual override fun getAddress(): Long {
        return this.a
    }

    actual override fun a(arg0: ByteArray?, arg1: Int, arg2: Int, arg3: Int) {
        if ((arg2 + arg3 > this.b) or (arg2 < 0) or ((arg0?.size ?: 0) < arg3 + arg1) or (arg1 < 0) or (arg0 == null) or (this.a == 0L)) {
            throw RuntimeException()
        }
        this.put(this.a, arg0, arg1, arg2, arg3)
    }

    actual override fun getSize(): Int {
        return this.b
    }

    /*protected*/ actual fun a(arg0: Long, arg1: Int) {
        this.b = arg1
        this.a = arg0
    }

    /*private*/ actual external fun get(arg0: Long, arg1: ByteArray?, arg2: Int, arg3: Int, arg4: Int)

    /*private*/ actual external fun put(arg0: Long, arg1: ByteArray?, arg2: Int, arg3: Int, arg4: Int)
}
